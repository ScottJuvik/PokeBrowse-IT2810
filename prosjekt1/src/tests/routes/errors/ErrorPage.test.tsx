import ErrorPage from '@/routes/errors/ErrorPage';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { afterEach, beforeEach, vi } from 'vitest';

const mockUseRouteError = vi.fn();

vi.mock('react-router-dom', async () => {
    const actual = await vi.importActual<typeof import('react-router-dom')>('react-router-dom');
    return {
        ...actual,
        useRouteError: () => mockUseRouteError(),
    };
});

beforeEach(() => {
    vi.spyOn(console, 'error').mockImplementation(() => {});
});

afterEach(() => {
    vi.restoreAllMocks();
    mockUseRouteError.mockReset();
});

describe('ErrorPage', () => {
    const renderComponent = () => {
        render(
            <MemoryRouter>
                <ErrorPage />
            </MemoryRouter>
        );

        return {
            image: screen.getByRole('img'),
            container: screen.getByTestId('errorPage-wrapper'),
        };
    };

    it('renders the error page with default message when no error is provided', () => {
        mockUseRouteError.mockReturnValue(undefined);

        const { image } = renderComponent();
        expect(image).toBeInTheDocument();
    });

    it('matches the snapshot', () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
    });
});
