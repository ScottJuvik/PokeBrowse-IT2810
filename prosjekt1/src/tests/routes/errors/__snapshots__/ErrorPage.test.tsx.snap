// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage > matches the snapshot 1`] = `
<main
  class="_errorContainer_ae0c75"
  data-testid="errorPage-wrapper"
>
  <img
    alt="Error Image"
    aria-label="Error image"
    src="/src/assets/gengar.png"
  />
  <h1>
    Uh-oh, Gengar's Mischief!
  </h1>
  <p
    class="_sorryText_ae0c75"
  >
    Sorry, an unexpected error has occured.
  </p>
  <p
    class="_prankText_ae0c75"
  >
    Gengar's ghostly pranks seem to have caused a disturbance in the web continuum.
  </p>
  <p
    class="_errorText_ae0c75"
  >
    Something went wrong!
  </p>
  <p
    class="_statusText_ae0c75"
  >
    <i>
      Mysterious forces at work...
    </i>
  </p>
</main>
`;
